feat: Remove autoheal functionality from rhsmcertd

jvlcek opened the pull request:
Card ID: CCT-723

Removed autoheal sub-command from rhsmcertd because autoheal is useless in pure SCA mode..

jvlcek commented:
@jirihnidek Here is my `draft` PR. Please take a look.

I have three questions:

1. How do I test these changes beyond running the pytests?
2. It looks to me like auto_attach_initial_delay , auto_attach_offset, and NEXT_AUTO_ATTACH_UPDATE_FILE can also be removed from src/daemons/rhsmcertd.c.  Do you know if that is correct?
3.  What should I do about the deprication message from man/rhsmcertd.8 ?
    https://github.com/candlepin/subscription-manager/blob/main/man/rhsmcertd.8#L103-L114

ptoscano commented on src/daemons/rhsmcertd.c:
```diff
@@ -86,33 +85,24 @@ static LOG_LEVEL log_level = DEFAULT_LOG_LEVEL;
 
 static GOptionEntry entries[] = {
     /* marked deprecated as of 02-19-2013, needs to be removed...? */
-    {"cert-interval", 0, 0, G_OPTION_ARG_INT, &arg_heal_interval_minutes,
+    {"cert-interval", 0, 0, G_OPTION_ARG_INT, &arg_cert_interval_minutes,
      N_("deprecated, see --cert-check-interval"),
      "MINUTES"},
```
`--cert-interval` is an alias for `--heal-interval`, so simply drop it; yes, the help text here is wrongly pointing to `--cert-check-interval`, even if the actual variable set is the one for `--heal-interval`

jirihnidek replied:
Yeah, If anything has been deprecated since 2013, then we can drop it now.

jvlcek replied:
Addressed by commit: [e4de3f4](https://github.com/candlepin/subscription-manager/pull/3455/commits/e4de3f402334d657c10ac83d2bafceeb5874d1fa)

ptoscano commented on src/daemons/rhsmcertd.c:
```diff
@@ -433,7 +422,7 @@ auto_register(gpointer data)
 
 static gboolean
-cert_check (gboolean heal)
+cert_check ()
```
this is C, so you need to specify `void` in case a function takes no parameters; actually, since this function is called as timeout (see `g_timeout_add()`), then the right parameter is `gpointer data`; to avoid warnings about the unused parameter, use `G_GNUC_UNUSED`:
```suggestion
cert_check (G_GNUC_UNUSED gpointer data)
```

ptoscano commented on src/daemons/rhsmcertd.c:
```diff
@@ -690,7 +659,6 @@ deprecated_arg_init_config (Config * config, int argc, char *argv[])
     }
 
     config->cert_interval_seconds = atoi (argv[1]) * 60;
-    config->heal_interval_seconds = atoi (argv[2]) * 60;
```
since this is removed you need to tweak the `argc` check above:
```c
    if (argc != 3) {
```
... which must be 2 now

jvlcek replied:
Addressed by commit [a8aedb6](https://github.com/candlepin/subscription-manager/pull/3455/commits/a8aedb6a6f8dacce552cdee4b664f6d72d62b924)

ptoscano reviewed (changes requested):
It generally seems good, there are few things to fix. Also drop an extra import, see the flake8 failure (it appears also as inline notes in the "files changed" view in github).

> It looks to me like auto_attach_initial_delay , auto_attach_offset, and NEXT_AUTO_ATTACH_UPDATE_FILE can also be removed from src/daemons/rhsmcertd.c. Do you know if that is correct?

Yes, it looks so.

> What should I do about the deprication message from man/rhsmcertd.8 ?

Edit it to mention only one parameter, according to the changes done here.

That feature looks something to remove as well, and I'd do that in a separate commit/PR.

jirihnidek commented on INSTALL.md:
```diff
@@ -100,7 +100,7 @@ Other versions or distributions may require some adaptation.
        PYTHONPATH=/path/to/subscription-manager/src \
        $(which python3) \
-       -m subscription_manager.scripts.rhsmcertd_worker --autoheal"
+       -m subscription_manager.scripts.rhsmcertd_worker --auto-register"
```
`--auto-register` exist, but it does not make sense to use it in this case.

ptoscano replied:
it's `--auto-registration` though ;-) and `--auto-register` is not a prefix of it

ptoscano replied:
oh the worker has it, I kept looking at rhsmcertd itself :facepalm:

jvlcek replied:
Resolved by commit: [f52c822](https://github.com/candlepin/subscription-manager/pull/3455/commits/f52c8227dc8593d2471446411fa4ec584f5a80d8)

jirihnidek commented on test/test_certmgr.py:
Do not forget on comment from Github action.

jvlcek replied:
Addressed by commit: [b939468](https://github.com/candlepin/subscription-manager/pull/3455/commits/b939468db15e30c314b9fd11fd5a8a17d6688819)

jirihnidek reviewed (commented):
I have same comments as Pino has.

jvlcek commented:
Thank you @ptoscano and @jirihnidek  for all of the help and patience.

I believe I have addressed all of the issues you've identified so far.

I will now squash all of the commits I created while resolving yourreview feedback.

ptoscano commented on man/rhsmcertd.8:
```diff
@@ -93,22 +87,22 @@ service rhsmcertd stop
 
 .SS DEPRECATED USAGE
-\fBrhsmcertd\fP used to allow the certificate and auto-attach intervals to be reset simply by passing two integers as arguments.
+\fBrhsmcertd\fP used to allow the certificate intervals to be reset simply by passing an integer as arguments.
```
nit: "as argument" (only one)

jvlcek replied:
Thank you @ptoscano, good catch. 
Address by commit [2e6f959](https://github.com/candlepin/subscription-manager/pull/3455/commits/2e6f959935887e122c70a56e156fb04e9c33a6aa)

ptoscano commented on src/daemons/rhsmcertd.c:
```diff
@@ -86,33 +84,21 @@ static LOG_LEVEL log_level = DEFAULT_LOG_LEVEL;
 };
 
 static GOptionEntry entries[] = {
     /* marked deprecated as of 02-19-2013, needs to be removed...? */
```
this comment belongs to the dropped `--cert-interval`, so must be removed too

jvlcek replied:
Thanks for catching this @ptoscano 
Addressed by commit: [9e27339](https://github.com/candlepin/subscription-manager/pull/3455/commits/9e2733943b276f08f981d680baf5a9c5ad6f81b7)

ptoscano commented on src/daemons/rhsmcertd.c:
```diff
@@ -474,11 +450,11 @@ static gboolean
 {
     struct CertCheckData *cert_data = data;
-    cert_check (cert_data->heal);
+    cert_check (data);
```
simply pass `NULL` here

jvlcek replied:
While I was on vacation it dawned on me that this change would be needed. I'm glad you pointed it out too.
Addressed by commit: [84272c3](https://github.com/candlepin/subscription-manager/pull/3455/commits/84272c302b6063aeb18b95b6150b068293489d81)

ptoscano reviewed (changes requested):
thanks for the updates, this seems better now; I'll give it a try later on

jvlcek commented:
Thank you for the feedback @ptoscano 
I will squash my comments are rebase.
